locust_plugins/jmeter_listener.py

```diff
@@ -0,0 +1,242 @@
+        """
+        self.results_file.write(self.row_delimiter.join(self.csv_results) + self.row_delimiter)
+        self.results_file.close()
+
+    def start_logging(self, user):
+        self.user_count += 1
+        self.user_name = user.__class__.__name__
+        if self.auto_log:
+            user.client.request = self._add_record(user.client.request)
+
+    def _add_record(self, func):
+        def wrapper(wrappedself, *args, **kwargs):
+            """
+            adds a result
+            """
+            timestamp = datetime.fromtimestamp(time()).strftime(self.timestamp_format)
+            name = kwargs["name"] if "name" in kwargs else "unknown"
+            response = func(wrappedself, *args, **kwargs)
+            if hasattr(response, "_manual_result"):
+                logging.info(
+                    "Found manually controlled result in '"
+                    + name
+                    + "' Change to manual logging to set pass/fail correctly"
+                )
+            try:
+                status_code = str(response.status_code)
+                thread_name = self.user_name
+                elapsed_time = str(round(response.elapsed.microseconds / 1000))
+                response_time = elapsed_time
+                response_length = str(len(response.text))
+                if response.ok:
+                    response_message = "OK"
+                    success = "true"
+                    exception = ""
+                else:
+                    response_message = "KO"
+                    success = "false"
+                    exception = str(response.reason)
+
+                binary_codecs = [
+                    "base64",
+                    "base_64",
+                    "bz2",
+                    "hex",
+                    "quopri",
+                    "quotedprintable",
+                    "quoted_printable",
+                    "uu",
+                    "zip",
+                    "zlib",
+                ]
+                data_type = "binary" if response.encoding in binary_codecs else "text"
+                bytes_sent = "0"
+                group_threads = str(self.user_count)
+                all_threads = str(self.runner.user_count)
+                latency = "0"
+                idle_time = "0"
+                connect = "0"
+
+                self.add_result(
+                    timestamp,
+                    response_time,
+                    name,
+                    status_code,
+                    response_message,
+                    thread_name,
+                    data_type,
+                    success,
+                    exception,
+                    response_length,
+                    bytes_sent,
+                    group_threads,
+                    all_threads,
+                    latency,
+                    idle_time,
+                    connect,
+                )
+            except:
+                logging.error("failed to log result")
```

Comment:
I'm hesitant to catch exceptions. I usually dont take the time, writing to a file should never fail, and if it does, your test result is usually screwed anyway :)

If you do want to catch the exception then you should at least print it.

Reply:
OK